terraform/s3.tf.json.template.py

```diff
@@ -19,6 +16,25 @@
                             "days": 1
                         }
                     }
+                },
+                "redirect_bucket": {
+                    "bucket": config.s3_redirect_bucket,
+                    "acl": "private",
+                    "website": {
+                        "index_document": "error.html",
+                        "error_document": "error.html"
+                    }
+                }
+            },
+            "aws_s3_bucket_object": {
+                "default_query_redirect": {
+                    "bucket": config.s3_redirect_bucket,
+                    "acl": "public-read",
+                    "key": "error.html",
+                    "content": " ",
+                    # TODO: Should not redirect to dev but there is no prod subdomain
+                    #  and preview doesn't have a 404 page
+                    "website_redirect": "https://dev.data.humancellatlas.org/404/"
```

Comment:
`data.humancellatlas.org/404/` doesn't exist right now so I put `dev`.  We could also use the 404 of the www subdomain or not redirect and show the s3 403 key not found page.

Reply:
I think the decision to redirect to the front-end was a mistake. We shouldn't assume which frontend is hitting us. Can we just return a generic, body-less 404 response?

Reply:
We can let s3 return a 404 by making the entire bucket public and not setting an error document.  The response body isn't empty though.

It will look like this:
```
404 Not Found

    Code: NoSuchKey
    Message: The specified key does not exist.
    Key: 404
    RequestId: ...
    HostId: ...
```

I can't find a way to return an empty 404 but I think this is fine.

Reply:
We can also keep the bucket private and return 403 if the object isn't found